crates/runtime/src/dialogue.rs

```diff
@@ -325,71 +309,90 @@ impl Dialogue {
     /// the node's source code. This header must be a space-separated list
     ///
     /// Returns [`None`] if the node is not present in the program.
-    pub fn get_tags_for_node(&self, node_name: &str) -> Option<impl Iterator<Item = &str>> {
-        self.get_node_logging_errors(node_name)
-            .map(|node| node.tags.iter().map(|s| s.as_str()))
+    pub fn get_tags_for_node(&self, node_name: &str) -> Option<Vec<String>> {
+        self.handler_safe_dialogue.get_tags_for_node(node_name)
     }
 
-    /// Gets a value indicating whether a specified node exists in the
-    /// Program.
+    /// Gets a value indicating whether a specified node exists in the Program.
     pub fn node_exists(&self, node_name: &str) -> bool {
-        // Not calling `get_node_logging_errors` because this method does not write errors when there are no nodes.
-        if let Some(program) = self.program() {
-            program.nodes.contains_key(node_name)
-        } else {
-            self.log_error_message
-                .call("Tried to call NodeExists, but no program has been loaded".to_owned());
-            false
-        }
+        self.handler_safe_dialogue.node_exists(node_name)
     }
 
-    pub fn analyse(&mut self) {
-        // ## Implementation notes
-        // It would be more ergonomic to not expose this and call it automatically.
-        // We should probs remove this from the API.
-        // Call it when running the first `continue_` after adding a program or something.
-        todo!()
+    /// Replaces all substitution markers in a text with the given
+    /// substitution list.
+    ///
+    /// This method replaces substitution markers - for example, `{0}`
```

Comment:
See the comment I already did. Unlucky everything is duplicated?

Reply:
#103 yeets this